tutorials/advanced-development/advanced-pieces-topics/adding-filters.md

```diff
@@ -1,40 +1,212 @@
+  {% endfor %}
+</ul>
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+### Showing the current state of the filter
+
+Usually you want to indicate the tag the user has already chosen. And, you want a way to remove the filter and to see the full results.
+
+How can we do that? Again, in `index.html`:
+
+{% code-tabs %}
+{% code-tabs-item title="lib/modules/people-pages/index.html" %}
+```markup
+{# Link to all the tags, adding a parameter to the query string #}
+<ul class="tag-filters">
+  {% for tag in data.piecesFilters.tags %}
+    <li>
+      {% if data.query.tags == tag.value %}
+        <a href="{{ data.url | build({ tags: '' }) }}" class="current">
+      {% else %}
+        <a href="{{ data.url | build({ tags: tag.value }) }}">
+      {% endif %}
+        {{ tag.label }}
+      </a>
+    </li>
+  {% endfor %}
+</ul>
 ```
 {% endcode-tabs-item %}
 {% endcode-tabs %}
 
-The same approach works for most types of schema fields, including joins. We do not recommend using it if the number of items in the dropdown will be very large. However, adding options to support filters that employ typeahead and avoid sending a large list of options to the browser is on our roadmap for the future.
+{% hint style="info" %}
+**"What's going on in this code?"** The current query string is automatically unpacked to `data.query` for you as an object. So just compare `data.query.tags` to the value of each of the choices.
+
+We add a `current` CSS class to the link to remove the current filter. It's up to you to style that; for instance, you might use an `::after` pseudo-element to add an "x."
+{% endhint %}
+
+### Filtering on joins and other schema field types
+
+Tags are the simplest example, but you can filter on most schema field types, notably including [`select`](../../schema-guide/schema-guide.md#select) fields and [`joinByOne`](/tutorials/schema-guide/schema-guide.md#joinByOne) or [`joinByArray`](/tutorials/schema-guide/schema-guide.md#joinByArray) fields.
+
+Add a filter on the `_jobs` schema field we saw earlier:
+
+{% code-tabs %}
+{% code-tabs-item title="lib/modules/people-pages/index.js" %}
+```javascript
+module.exports = {
+  piecesFilters: [
+    {
+      name: 'jobs'
+    }
+  ]
+}
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+{% hint style="info" %}
+"Why is the filter named `jobs`, even though the field is named `_jobs`?" It works like this: if we specify `_jobs` for the filter, then the value in the query string will be the `_id` property of the job. This works, and it is stable no matter what gets edited later. But it isn't pretty. If we remove the `_` from the filter name, the value in the query string will be the *slug* of the job, which is more user-friendly and good for SEO.
+
+However, keep in mind that if you change the slug someone's bookmarked links might break. So it's up to you whether to use `_jobs` (for the `_id`) or `jobs` (for the `slug`).
```

Comment:
This line can be changed to:

"Because Apostrophe records past URLs used to access each page and issues 'soft redirects,' broken links will be rare even if slugs change. So we recommend using `jobs` rather than `_jobs`."

Reply:
I think I but this in the right place, it was intended to replace the "However, keep in mind..." section?

Reply:
Yes. Slug changes don't typically cause a 404 anymore.